index.bs

```diff
@@ -0,0 +1,512 @@
+   * be called in a sitting space experience.
+   */
+  void resetPose();
+
+  /**
+   * z-depth defining the near plane of the eye view frustum
+   * enables mapping of values in the render target depth
+   * attachment to scene coordinates. Initially set to 0.01.
+   */
+  attribute double depthNear;
+  /**
+   * z-depth defining the far plane of the eye view frustum
+   * enables mapping of values in the render target depth
+   * attachment to scene coordinates. Initially set to 10000.0.
+   */
+  attribute double depthFar;
+
+  /**
+   * The callback passed to `requestAnimationFrame` will be called
+   * any time a new frame should be rendered. When the VRDisplay is
+   * presenting the callback will be called at the native refresh
+   * rate of the HMD. When not presenting this function acts
+   * identically to window.requestAnimationFrame. Content should make no
+   * assumptions of frame rate or vsync behavior as the HMD runs
+   * asynchronously from other displays and at differing refresh rates.
+   */
+  [Throws] long requestAnimationFrame(FrameRequestCallback callback);
+
+  /**
+   * Passing the value returned by `requestAnimationFrame` to
+   * `cancelAnimationFrame` will unregister the callback.
+   */
+  [Throws] void cancelAnimationFrame(long handle);
+
+  /**
+   * Begin presenting to the VRDisplay. Must be called in response to a user gesture.
+   * Repeat calls while already presenting will update the VRLayer being displayed.
+   */
+  Promise&lt;void&gt; requestPresent();
+
+  /**
+   * Stops presenting to the VRDisplay.
+   */
+  Promise&lt;void&gt; exitPresent();
+
+  /**
+   * Get the sources currently being presented.
+   */
+  sequence&lt;VRLayer&gt;? getLayers();
+
+  /**
+   * The VRLayer provided to the VRDisplay will be captured and presented
+   * in the HMD. Calling this function has the same effect on the source
+   * canvas as any other operation that uses its source image, and canvases
+   * created without preserveDrawingBuffer set to true will be cleared.
+   */
+  void submitFrame(optional VRPose pose);
+};
+</pre>
+
+### Attributes ### {#vrdisplay-attributes}
+
+<dfn attribute for="VRDisplay">isConnected</dfn>
+The {{isConnected}} attribute MUST return the {{VRDisplay}}'s connected state.
+
+<dfn attribute for="VRDisplay">isPresenting</dfn>
+The {{isPresenting}} attribute MUST return the {{VRDisplay}}'s presentation state.
+
+<dfn attribute for="VRDisplay">capabilities</dfn>
+The {{capabilities}} attribute MUST return the {{VRDisplay}}'s {{VRDisplayCapabilities}} object, a dictionary of capabilities describing the {{VRDisplay}}.
+
+
+## VRLayer ## {#vrlayer}
+
+The {{VRLayer}} interface is provided to a {{VRDisplay}} and presented in the HMD.
+
+<pre class="idl" id="vrlayer-dictionary">
+typedef (HTMLCanvasElement or
+         OffscreenCanvas) VRSource;
+
+dictionary VRLayer {
+  VRSource? source = null;
+
+  sequence&lt;float&gt; leftViewport;
+  sequence&lt;float&gt; rightViewport;
```

Comment:
These two arrays should be nullable and default to null.

I left a comment about this on the discussion doc, but I'd like to resolve if we still feel "viewport" is the right verbiage here, since viewports are typically given in pixels but we're using UVs here. Either we should accept pixel values here (which may be nice, because then you use the same values for both gl.viewport and this) or change the name to something like `leftBounds` and `rightBounds`.